web/src/filter.ts

```diff
@@ -189,8 +233,20 @@ function message_matches_search_term(message, operator, operand) {
     return true; // unknown operators return true (effectively ignored)
 }
 
+type Term = {
+    negated: boolean;
```

Comment:
It looks like `negated` isn’t always defined before `Filter.canonicalize_term`, which applies the default `negated = false`.

Reply:
Good catch, made it optional